### PR TITLE
Error handling and delete button

### PR DESCRIPTION
**Pull request by barkas90**
Added error handling and delete button

### CODE REVIEW

File: Desktop/AddGoalWindow.cs [hunk 1]
```diff
@@ -51,78 +53,138 @@ private void PopulateGoalListView()
...
+                descriptionBox.BackColor = Color.Red;
+            }
+
+            if (String.IsNullOrWhiteSpace(name))
+            {
+                goalNameBox.BackColor = Color.Red;
             }
         }
+
         private void button1_Click(object sender, EventArgs e)
```
Comment (imila254) [inline Desktop/AddGoalWindow.cs:141]: Button should have a specific name.

File: Desktop/AddGoalWindow.cs [hunk 2]
```diff
@@ -51,78 +53,138 @@ private void PopulateGoalListView()
...
+            var selectedTitle = listView1.Items[selectedId].Text;
+            listView1.Items.RemoveAt(selectedId);
+            db.RemoveGoal(selectedTitle);
+            UpdateGoalList();
+        }
+
+        private void listView1_SelectedIndexChanged(object sender, EventArgs e)
+        {
+
         }
```
Comment (imila254) [inline Desktop/AddGoalWindow.cs:188]: What is the purpose of this method?

**Review by imila254 — CHANGES_REQUESTED** [review]

File: Desktop/AddGoalWindow.cs (hunk 2 above)
Comment (barkas90) [reply Desktop/AddGoalWindow.cs:188]: This is automatically made by Visual studio, it has no purpose, i will delete it.

File: Desktop/AddGoalWindow.cs (hunk 1 above)
Comment (barkas90) [reply Desktop/AddGoalWindow.cs:141]: Thank you for noticing, i will rename it

**Review by imila254 — APPROVED** [review]

File: Desktop/AddGoalWindow.cs [hunk 3]
```diff
@@ -78,12 +81,19 @@ private void PopulateGoalListView(IEnumerable<Goal> GoalList)
...
             {
-                int money = (Decimal.ToInt32(goal.Amount)) / ((((DateTime)goal.GoalDate).Subtract(DateTime.UtcNow) / 7).Days);
+                int money;
+                if (((DateTime)goal.Deadlinedate).Subtract(DateTime.UtcNow).Days > 7)
+
+                    money = (Decimal.ToInt32(goal.Amount)) / ((((DateTime)goal.Deadlinedate).Subtract(DateTime.UtcNow) / 7).Days);
+
+                else
+                    money = (Decimal.ToInt32(goal.Amount));
+
```
Comment (AleksandrasMaliuginas) [inline Desktop/AddGoalWindow.cs:91]: Brackets have to be added.

File: Desktop/AddGoalWindow.cs [hunk 4]
```diff
@@ -110,41 +120,53 @@ private void goalAmount_KeyPress(object sender, KeyPressEventArgs e)
...
             if (String.IsNullOrWhiteSpace(details))
             {
-                goalMoney.BackColor = Color.Red;
+                descriptionBox.BackColor = Color.Red;
+            }
+
+            if (String.IsNullOrWhiteSpace(name))
+            {
+                goalNameBox.BackColor = Color.Red;
             }
```
Comment (AleksandrasMaliuginas) [inline Desktop/AddGoalWindow.cs:138]: Consider putting status colours into an Enum

Comment (barkas90) [reply Desktop/AddGoalWindow.cs:138]: I don't think Enum is useful when using only one color

Comment (AleksandrasMaliuginas) [reply Desktop/AddGoalWindow.cs:141]: Oh, I missed that, you are right!

**Review by AleksandrasMaliuginas — APPROVED** [review]

**Pull request merged**